app/src/main/java/com/sun/ise/ui/login/LoginActivity.kt

```diff
@@ -1,13 +1,58 @@
 package com.sun.ise.ui.login
 
+import android.content.Intent
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProviders
 import com.sun.ise.R
+import com.sun.ise.data.local.LocalDataSource
+import com.sun.ise.data.remote.RemoteDataSource
+import com.sun.ise.data.remote.RetrofitService
+import com.sun.ise.data.api.UserApi
+import com.sun.ise.data.repository.UserRepository
+import com.sun.ise.ui.main.MainActivity
+import com.sun.ise.util.*
+import kotlinx.android.synthetic.main.activity_login.*
 
 class LoginActivity : AppCompatActivity() {
 
+    private val TAG = LoginActivity::class.java.simpleName
+    private val retrofit by lazy {
+        RetrofitService.createService(UserApi::class.java)
+    }
+    private val loginViewModel: LoginViewModel by lazy {
+        ViewModelProviders.of(
+            this,
+            ViewModelUtil.viewModelFactory {
+                LoginViewModel(
+                    UserRepository(
+                        LocalDataSource(this),
```

Comment:
```suggestion
                        LocalDataSource(application),
```
nếu tương tác với data không cần activity, thường thi Data source sẽ được khởi tạo với application hơn.